crates/gen-host-js/src/lib.rs

```diff
@@ -587,28 +619,31 @@ impl Js {
                         ptr = realloc(ptr, allocLen, 1, writtenTotal);
                     utf8EncodedLen = writtenTotal;
                     return ptr;
-                }
+                }\n
```

Comment:
FWIW I realize these are added from making the generated code look better, but from a maintainability point of view this is really hard to work with since there's just sort of random `\n` markers all sprinkled about without any particular rhyme or reason. Without tests whatever output is precisely done today is extremely likely to regress or change in the future. Personally I find that the readability is hindered with these sorts of markers because it makes me wonder why they're here at all which I have to actively combat with an understanding that it's just for generated code appearance at some point in the past.

Reply:
The rule I'm using for output is quite simple - in the generated output, top-level declaration blocks get double spacing. In generated functions, there is no unnecessary whitespace. No spaces more than a double line spacing. Then no leading whitespace, and trailing whitespace at the end of the file. An output linter rule could be configured to catch this I think.

Reply:
That all makes sense but doesn't really address my point. I don't particularly care about the actual output of the generated code that much, the problem here is the source code itself. The source code has all these little changes sprinkled throughout it where the rhyme-and-reason is only apparent if you follow it through all the way to the end. Personally I don't think that it's the right balance since generated code is generated code and the source code here is pretty important to be readable.

I would prefer that, if possible, issues like this are fixed at the `Source` level to avoid the need to go through contortions at source code layer written in Rust.

Reply:
I will check more carefully before commenting - turns out your intuition was exactly right in that this newline was incorrect!